cmd/tailscale/cli/update.go

```diff
@@ -333,6 +333,65 @@ func (up *updater) updateMacSys() error {
 	return errors.New("The 'update' command is not yet implemented on macOS.")
 }
 
+func (up *updater) updateMacAppStore() error {
+	// It turns out that `--schedule` only prints "Automatic checking for updates
+	// is turned on" if *checking* is turned on — even if *installing* updates is
+	// turned off. So a result of "on" isn't as good as we might hope.
+	//
+	// That said, a result of "off" is note-worthy.
+	out, err := exec.Command("softwareupdate", "--schedule").CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("can't check App Store update schedule: %w, output: %q", err, string(out))
+	}
+	const off = "Automatic checking for updates is turned off"
+	if strings.Contains(string(out), off) {
+		fmt.Fprintln(os.Stderr, "NOTE: Automatic checking for macOS/App Store updates is turned off. You can change this setting in System Settings (search for ‘update’).")
+	}
+
+	out, err = exec.Command("softwareupdate", "--list").CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("can't check App Store for available updates: %w, output: %q", err, string(out))
+	}
+
+	newTailscale := parseSoftwareupdateList(out)
+	if newTailscale == "" {
+		fmt.Println("no Tailscale update available")
+		return nil
+	}
+
+	// parseSoftwareupdateList guarantees the format of the version string, but
+	// check and document it here, too. A case could be made for `!= 2`, not just
+	// `< 2`, as well.
+	newTailscaleParts := strings.Split(newTailscale, "-")
+	if len(newTailscaleParts) < 2 {
+		return fmt.Errorf(`Tailscale version from the App Store has unexpected format: %q, expect "Tailscale-x.y.z"`, newTailscale)
+	}
```

Comment:
Thinking about it more, this could just be `newTailscaleVer := strings.TrimPrefix(newTailscale, "Tailscale-")` :sweat_smile:

Reply:
Thanks! :)